homeassistant/components/picnic/todo.py

```diff
@@ -73,3 +81,18 @@ def todo_items(self) -> list[TodoItem] | None:
                 )
 
         return items
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add item to shopping cart."""
+        product_id = await self.hass.async_add_executor_job(
+            product_search, self.coordinator.picnic_api_client, item.summary
+        )
+
+        if not product_id:
+            raise ValueError("No product found or no product ID given")
```

Comment:
Raise ServiceValidationError for bad user input and HomeAssistantError for other problems in service calls.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/